src/docdb/nodes.ts

```diff
@@ -0,0 +1,186 @@
+import * as vscode from 'vscode';
+import * as vm from 'vm';
+import * as path from 'path';
+import { EventEmitter, Event, Command } from 'vscode';
+import { AzureAccount } from '../azure-account.api';
+import { INode, ErrorNode } from '../nodes';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import docDBModels = require("azure-arm-documentdb/lib/models");
+import DocumentdbManagementClient = require("azure-arm-documentdb");
+import { MongoDatabaseNode } from '../mongo/nodes';
+import {MongoCommands} from '../mongo/commands'
+var DocumentDBConnectionClient = require("documentdb").DocumentClient;
+
+
+export interface DocDBCommand {
+	range: vscode.Range;
+	text: string;
+	collection?: string;
+	name: string;
+	arguments?: string;
+}
+
+export interface IDocDBServer extends INode {
+	getPrimaryMasterKey(): string;
+	getEndpoint(): string;
+}
+
+export class DocDBServerNode implements IDocDBServer {
+	readonly contextValue: string = "DocDBServer";
+	readonly label: string;
+
+	readonly collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+
+	constructor(readonly _primaryMasterKey: string, readonly id: string, private readonly _endpoint: string) {
+		this.label = id;
+	}
+
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'light', 'DataServer.svg'),
+			dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'dark', 'DataServer.svg')
+		};
+	}
+
+	getPrimaryMasterKey(): string {
+		return this._primaryMasterKey;
+	}
+
+	getEndpoint(): string {
+		return this._endpoint;
+	}
+
+	getChildren(): Promise<INode[]> {
+		//return new Promise<DocDBServerNode[]>;
+		let client = new DocumentDBConnectionClient(this.getEndpoint(), this.getPrimaryMasterKey());
+		return this.getDocDBDatabaseNodes(client, this);
+	}
+
+	async listDatabases(client): Promise<any[]> {
+		let databases = await client.readDatabases();
+		return await new Promise<any[]>((resolve, reject) => {
+			databases.toArray( (err, dbs: Array<Object>) => err ? reject(err) : resolve(dbs) );
+		});
+	}
+
+	async getDocDBDatabaseNodes(client, DocDBServerNodeInstance): Promise<INode[]> {
+		let databases;
+		try {
```

Comment:
Remove the try/catch

As dicussed offline - we want errors to be consistenly displayed by VS Code and we want the error to bubble up so we track it in telemetry